dask/array/tests/test_reductions.py

```diff
@@ -809,3 +809,28 @@ def test_nan_func_does_not_warn(func):
     with pytest.warns(None) as rec:
         getattr(da, func)(d).compute()
     assert not rec  # did not warn
+
+
+@pytest.mark.parametrize("chunks", list(permutations(((2, 1) * 8, (3,) * 8, (6,) * 4))))
+@pytest.mark.parametrize("split_every", [2, 4, 8, 16])
+@pytest.mark.parametrize(
+    "axes", list(permutations((0, 1, 2), 2)) + list(permutations((0, 1, 2)))
+)
+def test_chunk_structure_independence(axes, split_every, chunks):
+    # Reducing an array should not depend on its chunk-structure!!!
+    # See Issue #8541: https://github.com/dask/dask/issues/8541
+    shape = tuple(np.sum(s) for s in chunks)
+    np_array = np.arange(np.prod(shape)).reshape(*shape)
+    x = da.from_array(np_array, chunks=chunks)
+    reduced_x = da.reduction(
+        x,
+        lambda x, axis, keepdims: x,
+        lambda x, axis, keepdims: x,
+        keepdims=True,
+        axis=axes,
+        split_every=split_every,
+        dtype=x.dtype,
+        meta=x._meta,
+    )
+    reduced_x.compute_chunk_sizes()
+    assert_eq(reduced_x, np_array)
```

Comment:
`compute_chunk_sizes` just computes and sets the chunks, negating the purpose of checking them in `assert_eq`. Since your reduction doesn't actually reduce anything (and thus doesn't have a fixed output shape in any dimension), there's not a good way to pass the chunk information through `reduction`. Instead, we can just ignore checking the chunks in `assert_eq` (and add a comment that the chunks are intentionally wrong for this test case).

```suggestion
    # Skip checking chunks/shape; they're incorrect since this reduction doesn't actually
    # reduce anything. This isn't a bug, it's just due to the test case. 
    assert_eq(reduced_x, np_array, check_chunks=False, check_shape=False)
```

Reply:
Good point.  I'll do that.